### ISSUE TITLE
USER-DPD rx should not hardcode indices into atom->dname and atom->dvector arrays

### ISSUE THREAD

**Issue by akohlmey** [lammps/lammps#131]
The various "rx" components in USER-DPD has an arbitrary requirement, that no other feature in LAMMPS may use fix property/atom. This is against the modular design of LAMMPS and not really needed. For the same reason, there is no need to define two fix property/atom instances; one should be enough. all that would be needed to achieve the same features, would be generating a table (or map) that identifies which specific index in the atom->dname[] array matches with the desired property and then access them accordingly.

**Comment by timattox** [comment]
I will look into this.  Thanks for finding and pointing out the problem.

**Comment by akohlmey** [comment]
yeah, i was looking into this while working on the weighted load balancing, where i initially thought, that using fix property/atom would be the right way to store the per-atom weight parameter and was looking at other uses of it. but since property/atom by construction has a global namespace, things are a bit complicated and i got stuck trying to resolve access conflicts between fix balance and the balance command. in the end, i went for a simpler solution and used fix store (which is only internal and does not support communicating properties to ghost atoms) instead, which is sufficient in my case. fix property/atom is beneficial, if you have properties that you want to read in from a data file and restart as well. you just need to give them uniquely identifiable names, e.g. DPD_RX_spec_1  for species/type 1 and so on and then look them up via atom->find_custom(). to speed up the process, you could simply store those indices in an atom->ntypes size array. if fix store is sufficient, you simple need to give it a suitable fix-ID and then look up the fix and then refer to the stored data.

HTH,
      axel.
